drivers/modem/modem_cellular.c

```diff
@@ -1560,6 +1581,32 @@ static int modem_cellular_init(const struct device *dev)
 							&dlci2_config);
 	}
 
+	{
+		const struct modem_cmux_dlci_config dlci3_config = {
+			.dlci_address = 3,
+			.receive_buf = data->dlci2_receive_buf,
+			.receive_buf_size = ARRAY_SIZE(data->dlci3_receive_buf),
+		};
+
+		data->dlci3_pipe = modem_cmux_dlci_init(&data->cmux, &data->dlci3,
+							&dlci3_config);
+
+		modem_pipelink_init(config->shell_pipelink, data->dlci3_pipe);
+	}
+
+	{
+		const struct modem_cmux_dlci_config dlci4_config = {
+			.dlci_address = 4,
+			.receive_buf = data->dlci4_receive_buf,
+			.receive_buf_size = ARRAY_SIZE(data->dlci4_receive_buf),
+		};
+
+		data->dlci4_pipe = modem_cmux_dlci_init(&data->cmux, &data->dlci4,
+							&dlci4_config);
+
+		modem_pipelink_init(config->shell_pipelink, data->dlci4_pipe);
```

Comment:
```suggestion
		modem_pipelink_init(config->gnss_pipelink, data->dlci4_pipe);
```
Also, the DLCI initialization could be put into a helper function.